### ISSUE TITLE
hack/test-cmd.sh should passes its own kubectl context

### ISSUE THREAD

**Issue by caesarxuchao** [kubernetes/kubernetes#9167]
Currently, if the kubectl's context is using a namespace other than "default", hack/test-cmd.sh will fail at this line:
```
kube::test::get_object_assert services "{{range.items}}{{$id_field}}:{{end}}" 'kubernetes:kubernetes-ro:'
```
Error message:
```
FAIL!
Get services {{range.items}}{{.metadata.name}}:{{end}}
  Expected: kubernetes:kubernetes-ro:
  Got:  
```
this is because these services are in namespace default.

Usually this is not a problem because test-cmd.sh runs on Shippable and Travis, but the script will be more robust if it passes on its own context when calls kubectl.

@lavalamp @bgrant0607

**Comment by lavalamp** [comment]
To rephrase: hack/test-cmd.sh (and probably the e2e tests, too) have a non-obvious dependency on ```~/.kube/config``` which can cause them to behave differently locally and in the CI systems.

**Comment by caesarxuchao** [comment]
Thanks @lavalamp

**Comment by caesarxuchao** [comment]
hack/test-update-storage-object.sh also depends on ~/.kube/config

**Comment by ixdy** [comment]
Is this still an issue?

**Comment by caesarxuchao** [comment]
Yes, but it's almost innocuous. I'll send a fix later.

**Comment by caesarxuchao** [comment]
Actually it's fixed by #12009.
